pkg/binding/buffering/binary_buffer_message.go

```diff
@@ -21,35 +20,52 @@ type binaryBufferedMessage struct {
 	body       *bytebufferpool.ByteBuffer
 }
 
+func (m *binaryBufferedMessage) Init() error {
```

Comment:
for all these methods, I have found passing ctx helps for a bunch of things like tracing, metrics, and a hook to extend singleton state into methods that don't have to have the api change.

Reply:
Apart the `Init()` and `End()`, I don't see the reason to pass a `Context` in `BinaryEncoder` and `StructuredEncoder` interface. But I see the reason to do that in `Encode`, `RunDirectEncoders`, `CopyMessage` and `BufferMessage`

Reply:
wdyt https://github.com/cloudevents/sdk-go/pull/302/commits/5e760fa2f6981a222cc1f58983e95df18b00ad54 ?